src/web/rustdoc.rs

```diff
@@ -367,6 +368,10 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
     // Get the latest version of the crate
     let latest_version = latest_release.version.to_string();
     let is_latest_version = latest_version == version;
+    let is_prerelease = semver::Version::parse(&version)
+        // should be impossible unless there is a bug in match_version
+        .map_err(|_| Nope::InternalServerError)?
+        .is_prerelease();
```

Comment:
I don't think this is true, we allow exact matches before checking semver: https://github.com/rust-lang/docs.rs/blob/fb06f95e4a8537fa321c3b192a73ffa6dbd68e97/src/web/mod.rs#L333
```suggestion
    // If this is an exact match (not a valid semver), assume it's not a stable version.
    let is_prerelease = semver::Version::parse(&version).unwrap_or(true);
```

Reply:
Oh, you're right. But the suggested change won't work, because then `GET /sqlx/0.3` would be treated as a pre-release although the semver compatible version 0.3.5 would be returned.  
I thought just parsing the Version here would be the least invasive change, but it seems like I'll have to get the actual version returned from the DB for the release serverd.

Reply:
Mhh, i looked at it again. The `version` is actually guaranteed to be semver compatible, because in the case of an exact match the version is semver compatible (if not, the db contains an invalid semver version) and in the case of an inexact match `rustdoc_html_server_handler` will redirect to the exact version.
https://github.com/rust-lang/docs.rs/blob/c444b639dd489a8985426217feea5b1ec8d97c2c/src/web/rustdoc.rs#L303-L310

Reply:
I don't follow, `"0.3"` is a valid semver and would be parsed successfully.

Reply:
> if not, the db contains an invalid semver version

Hmm ... I don't think we enforce semantic versioning in the database, we depend on cargo to do it for us. In principle a 500 is probably the right thing to do, but it seems a shame to return that when it's so easy to return the page with a different text instead.

Reply:
Would you rather add a new `Nope` variant for this case? I feel like in this case it should be fine to just return a 500. We do the same in the `match_version` func https://github.com/rust-lang/docs.rs/blob/c444b639dd489a8985426217feea5b1ec8d97c2c/src/web/mod.rs#L348-L360

Reply:
Hmm, the `Nope` error would just turn into 'InternalServerError', right? I think that would be useful after #1046 to print the log message automatically (in the `from` impl) but other than that I don't think it buys us much.